submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+        } else {
+          currentGender.closest('.user-card').classList.remove('hidden');
+        }
+      });
+    }
+  }
+
+  insertSortedProfiles(sortedCards, param) {
+    sortedCards.forEach(card =>
+      this.profilesContainer.insertBefore(card, this.profilesContainer[param])
+    );
+  }
+
+  sortBySelector(selectorName) {
+    const cards = document.querySelectorAll('.user-card');
+    if (selectorName === 'all') {
+      return Array.from(cards).sort((a, b) => a.id.localeCompare(b.id));
+    }
+    return Array.from(cards).sort((a, b) =>
+      a
+        .querySelector(selectorName)
+        .textContent.localeCompare(b.querySelector(selectorName).textContent)
+    );
+  }
+
+  sortByContent(sortSelector, sortType, sortTypes) {
+    const sortedCards = this.sortBySelector(sortSelector);
+    const typeOfSort = {
+      [sortTypes.type1]: () =>
+        this.insertSortedProfiles(sortedCards, 'lastChild'),
+      [sortTypes.type2]: () =>
+        this.insertSortedProfiles(sortedCards, 'firstChild'),
+      [sortTypes.type3]: () =>
+        this.insertSortedProfiles(this.sortBySelector('all'), 'lastChild'),
+      default: () => false,
+    };
+    return (typeOfSort[sortType] || typeOfSort['default'])();
+  }
+}
+
+class Api {
+  constructor() {
+    this.apiURL = `https://randomuser.me/api/`;
+    this.errorMessage = `Uh oh, something has gone wrong. Please tweet us @randomapi about the issue. Thank you.`;
+  }
+
+  async getData(pageNum = 1) {
+    try {
+      const response = await fetch(
+        this.apiURL + `?page=${pageNum}&results=10&seed=abc`
+      );
+      if (!response.ok) throw new Error(errorMessage);
+      const data = await response.json();
+      new Render(data.results, data.info).getProfiles();
+      document.querySelector('.loading').classList.remove('active');
+    } catch (err) {
+      console.error(err);
+      alert(err);
+    }
+  }
+}
+
+class Observer {
+  constructor() {
+    this.startPage = 1;
+    this.loading = () => {
+      if (
+        window.scrollY + window.innerHeight >=
+        document.documentElement.scrollHeight - 5
+      ) {
+        document.querySelector('.loading').classList.add('active');
+        new Api().getData(++this.startPage);
+      }
+    };
+    document.querySelector('#menu').addEventListener('click', () => {
+      document.querySelector('#filters').classList.toggle('active');
+    });
+  }
+
+  loadDataByScroll(loadingStop) {
+    if (loadingStop) {
+      window.removeEventListener('scroll', this.loading);
+      return;
+    }
+    window.addEventListener('scroll', this.loading);
+  }
+}
+
+new Api().getData();
+new Filter();
+const loadMoreData = new Observer();
+loadMoreData.loadDataByScroll();
+setTimeout(() => {
+  document.querySelector('#preload').remove();
+  document.querySelector('body').style.setProperty('--scroll', 'auto');
+  document.querySelector('.wrapper').classList.remove('hidden');
+}, 7000);
```

Comment:
And what will be if 7 seconds are pass and data will not be loaded? :)

Do not use such approaches - you don't know how much time downloading of data will take. You can start load and `then` do what you need.

Reply:
This chunk of code responsibility for preload animation only. I need 7 seconds for right rendering text and image that removed this from page after render. Downloading of data is parallel with animation in this case. There was not desire to hide downloading, it was just for render animation at the beginning, before we see database of users.
Maybe I need to swap this parts of code. First might be rendering of animation and than downloading data (now is opposite) ?

Reply:
Looks like this is problem with names.

If you need this 7 seconds only because of animation you should:

1) save 7000 into constant with proper name
2) Separate animation and loading of data into separate functions
3) Call them inside function which will initialise your application

Reply:
Agreed, I separated this to another ```class Animation```

Reply:
And why it should be class? 🤔 Class is overkill for this. You can just use function

Reply:
Ok. Now it is ```function animation()```. That was just my desire to follow one code style. Like 'only classes used, no mix with functions'